test/NFTOpt.tests.ts

```diff
@@ -177,10 +177,18 @@ describe("NFTOpt Tests", function () {
 
     describe("withdrawOptionRequest", function () {
         it("should test that method can be called", async function () {
-            expect(NFTOptCTR.connect(buyer)
-                            .withdrawOptionRequest(0))
-            .to.not.throw;
-        })
+            expect(
+                NFTOptCTR.connect(buyer).withdrawOptionRequest(0)
+            ).to.not.throw;
+        });
+
+        it("should fail when caller is not the buyer", async function () {
```

Comment:
this is wrong, it doesn't fail as expected

Reply:
Didn't see the same thing when I ran the test, but will check it ou
![Screenshot 2022-05-22 123235](https://user-images.githubusercontent.com/2546321/169705891-4dc1daa7-cd4e-43af-a6f9-a444c2d54e08.png)
t

Reply:
Point #1:
you are expecting the method to _fail when the caller is not the buyer_
in the method you are not checking what you say you are checking
what you _are "checking"_ is if the seller can withdraw an non-existing option request (id 0)

Point #2:
expect() returns a promise;
you are not waiting on the promise, therefore the method "passes"
you do "expect" the call not to throw, yet it throws (because the you connect seller account and option(0) doesn't exist, so option(0).buyer = address(0) and that's not the seller account you connected). 
you don't know it fails because there's no `await expect`, and your test 'passes'

this does something else entirely than what you think it does :)

Reply:
Thanks again for the clarification.  As briefly mentioned yesterday I was copying some of the existing test cases in an attempt to rapidly get something out when I was still fuzzy on Typescript and HH.